tests/src/main/java/io/apicurio/tests/RegistryFacade.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apicurio.tests;
+
+import io.apicurio.tests.executor.Exec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.Future;
+
+public class RegistryFacade {
+    static final Logger LOGGER = LoggerFactory.getLogger(RegistryFacade.class);
+
+    public static final String DEFAULT_REGISTRY_JAR_PATH = "../app/target/apicurio-registry-app-1.0.2-SNAPSHOT-runner.jar";
+    public static final String DEFAULT_REGISTRY_PORT = "8080";
+    public static final String DEFAULT_REGISTRY_URL = "localhost";
+
+    public static final String REGISTRY_JAR_PATH = System.getenv().getOrDefault("REGISTRY_JAR_PATH", DEFAULT_REGISTRY_JAR_PATH);
+    public static final String REGISTRY_PORT = System.getenv().getOrDefault("REGISTRY_PORT", DEFAULT_REGISTRY_PORT);
+    public static final String REGISTRY_URL = System.getenv().getOrDefault("REGISTRY_URL", DEFAULT_REGISTRY_URL);
+    public static final String EXTERNAL_REGISTRY = System.getenv().getOrDefault("EXTERNAL_REGISTRY", "");
+
+    private Exec executor = new Exec();
+
+    private Future<Boolean> registries;
```

Comment:
Registries, as in many of them? There should be only one ...
With different Storage type -- in-memory (as default), JPA, Kafka, Streams, Infinispan, ...